### ISSUE TITLE
Bug in division

### ISSUE THREAD

**Issue by mohamed-barakat** [Singular/Sources#983]
With the latest Singular 2bfe11f5e I get the following error

```
> ring r= 0,(x),ds;
> matrix M[2][2]=[1,0,0,0];
> print(M);
1,0,
0,0
> matrix NN[2][1]=[1,0];
> module N=std(NN);
> list l=division(M,N);
> matrix U = l[3];
> print(U); // this matrix used to be a 2x2 matrix in previous versions of Singular
1
> matrix(M)*U;
   ? matrix size not compatible(2x2, 1x1) in *
   ? error occurred in or before STDIN line 7: `matrix(M)*U;`
   ? wrong type declaration. type 'help matrix;'
```
According to the documentation of [`division`](https://www.singular.uni-kl.de/Manual/4-1-2/sing_233.htm#SEC273) the last operation should work. The problem is that the matrix `U` has the wrong dimension. It used to be the 2x2 identity matrix, e.g. in version e0abce6.

```
Singular for x86_64-Darwin version 4.1.2 (4125, 64 bit) Mar 30 2020 22:09:34 #2bfe11f5e
with
	GMP(6.2.0),NTL(11.4.3),FLINT(2.5.2),factory(4.1.1),
	omalloc,static readline(8),Plural,DBM,
	dynamic modules,dynamic p_Procs,OM_NDEBUG,SING_NDEBUG,
	CC:OPTIMIZE,GenericAdd,TableMult,no invTable,
	eigenvalues,Gauss-Manin system,random=1585599111
built-in modules: {}
AC_CONFIGURE_ARGS =  '--prefix=/Users/xx/software/Singular/Singular/spielwiese/2bfe11f' 'LDFLAGS=-L/usr/local/opt/readline/lib' 'CPPFLAGS=-I/usr/local/opt/readline/include' --enable-omalloc  OMALLOC_LIBS='/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/omalloc/libomalloc.la' OMALLOC_INCLUDES='-I/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/../../../spielwiese -I/Users/xx/software/Singular/Singular/spielwiese/2bfe11f' --with-Singular RESOURCES_LIBS='/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/resources/libsingular_resources.la' RESOURCES_INCLUDES='-I/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/../../../spielwiese ' FACTORY_LIBS='/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/factory/libfactory.la' FACTORY_INCLUDES='-I/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/../../../spielwiese -I/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/../../../spielwiese/factory/include -I/Users/xx/software/Singular/Singular/spielwiese/2bfe11f -I/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/factory/include',
CC = gcc,FLAGS : -g  -pipe -fno-common -g0 -O3 -Wno-unused-function -Wno-trigraphs -Wno-unused-parameter -Wunknown-pragmas -Wno-unused-variable -fomit-frame-pointer -fwrapv -fvisibility=default -finline-functions -fno-exceptions -fno-threadsafe-statics -funroll-loops -fno-delete-null-pointer-checks -Qunused-arguments -pthread  -pthread,
CXX = g++,FLAGS :  -pipe -fno-common -g0 -O3 -Wno-unused-function -Wno-trigraphs -Wno-unused-parameter -Wunknown-pragmas -Wno-unused-variable -fomit-frame-pointer -fwrapv -fvisibility=default -finline-functions -fno-exceptions -fno-threadsafe-statics -funroll-loops -fno-delete-null-pointer-checks -fno-rtti -Qunused-arguments -pthread,
DEFS : ,CPPFLAGS : -I/usr/local/opt/readline/include,
LDFLAGS : -L/usr/local/opt/readline/lib -pipe -fno-common -g0 -O3 -Wno-unused-function -Wno-trigraphs -Wno-unused-parameter -Wunknown-pragmas -Wno-unused-variable -fomit-frame-pointer -fwrapv -fvisibility=default -finline-functions -fno-exceptions -fno-threadsafe-statics -funroll-loops -fno-delete-null-pointer-checks -Qunused-arguments  -rdynamic -flat_namespace -Wl,-bind_at_load -Wl,-undefined,dynamic_lookup,LIBS : -lreadline -lncurses -lmpfr -lgmp    (ver: 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.32.29))
argv[0]   :	Singular
SearchPath:	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../share/singular/LIB:/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../share/factory:/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../factory:/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../libexec/singular/MOD:/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/libexec/singular/MOD:/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin
Singular  :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/Singular
BinDir    :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin
ProcDir   :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../libexec/singular/MOD:/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/libexec/singular/MOD
RootDir   :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/..
DataDir   :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../share
DefaultDir:	/Users/xx/software/Singular/Singular/spielwiese/2bfe11f
InfoFile  :
IdxFile   :
HtmlDir   :
ManualUrl :	https://www.singular.uni-kl.de/Manual/
ExDir     :
Path      :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin:/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../libexec/singular/MOD:/Users/xx/software/Singular/Singular/spielwiese/2bfe11f/libexec/singular/MOD:/usr/local/opt/openssl/bin:/Applications/Macaulay2-1.14/bin:/Users/xx/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/MacGPG2/bin:/Library/TeX/texbin:/Applications/Magma:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/local/sbin:/Users/xx/Library/Python/2.7/bin
emacs     :	/usr/local/bin/../Cellar/emacs/26.3/bin/emacs-26.3
xemacs    :
SingularEmacs:	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/Singular
EmacsLoad :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../share/singular/emacs/.emacs-singular
EmacsDir  :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../share/singular/emacs
SingularXterm:	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/Singular
xterm     :	/opt/X11/bin/xterm
EmacsDir  :	/Users/xx/bin/../software/Singular/Singular/spielwiese/current/bin/../share/singular/emacs
Available HelpBrowsers: mac-www, safari-www, dummy, emacs,
Current HelpBrowser: mac-www
                     SINGULAR                                 /  Development
 A Computer Algebra System for Polynomial Computations       /   version 4.1.2
                                                           0<
 by: W. Decker, G.-M. Greuel, G. Pfister, H. Schoenemann     \   Mar 2020
FB Mathematik der Universitaet, D-67653 Kaiserslautern        \```

**Comment by mohamed-barakat** [comment]
@zickgraf: This is why the `homalg` currently fails.

**Comment by mohamed-barakat** [comment]
Fixed with 04929b74.

Thanks @hannes14.

It would be great to have a new release of Singular with the commit included.
